### DISCUSSION TITLE
Need a help for an "sometimes" Error in Forge and want to switch to Forge-UI from linked A-1111.

### DISCUSSION THREAD

**Discussion by cxbignekoc** [lllyasviel/stable-diffusion-webui-forge#1957] [Q&A]
Hi there,

I was used A1111 before switching to Forge, i like Forge so i want to stay.
Forge Running fine on Win_10, i5_14600K, RTX_4070_Ti_Super_16GB VRAM, 64_GB_RAM.

**Forge Version:**
`version: f2.0.1v1.10.1-previous-545-gf5190349  •  python: 3.10.11  •  torch: 2.1.2+cu121  •  xformers: 0.0.23.post1  •  gradio: 4.40.0  •  checkpoint: 321b1198d8` + diffusers: 0.29.2 + transformers: 4.44.0

All my Loras, Models, some Wildcards, VAE (and some more stuff?) are in the "stable-diffusion-webui" aka "a1111".

My ForgeUI folder "webui_forge_cu121_torch231" is what i actually use.

**I linked it by editing the webui-user.bat**:

```
@echo off

set PYTHON=C:/Users/user/AppData/Local/Programs/Python/Python310/python.exe
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --xformers

@REM Uncomment following code to reference an existing A1111 checkout.
 set A1111_HOME=C:/Users/user/Desktop/stable-diffusion-webui
 set VENV_DIR=%A1111_HOME%/venv
 set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% ^
 --ckpt-dir %A1111_HOME%/models/Stable-diffusion ^
 --hypernetwork-dir %A1111_HOME%/models/hypernetworks ^
 --embeddings-dir %A1111_HOME%/embeddings ^
 --lora-dir %A1111_HOME%/models/Lora

call webui.bat
```

1. I want to switch to Forge. And need to Edit the web-ui? The Reason for that was just a Youtube Video where this was explained for my Situation. I just wanted try Forge.
2. Can i just Rename the Folder after switching "webui_forge_cu121_torch231", because "update.bat" actualized all in the that Folder, right?
3. How to uninstall "stable-diffusion-webui" aka "a1111" or just delete Folder?

=========

The ERROR how i get it was in a Download Session - a second try finished the Updates - so i think its secondary or FYI:

1. Go in "check for updates", disable NONE but "sd-webui-controlnet" is unchecked?
2. Apply and Restart UI
3. ERROR's a lot of

```
You are running torch 2.1.2+cu121.
The program is tested to work with torch 2.3.1.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
```

==============================================================================

```

You are running xformers 0.0.23.post1.
The program is tested to work with xformers 0.0.27.
To reinstall the desired version, run with commandline flag --reinstall-xformers.

Use --skip-version-check commandline argument to disable this check.
```

================================================================================

```
ControlNet preprocessor location: C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\models\ControlNetPreprocessor
*** Error loading script: BatchParams.py
    Traceback (most recent call last):
      File "C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\modules\scripts.py", line 525, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\modules\script_loading.py", line 13, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\BatchParams.py", line 7, in <module>
        from scripts.Utils import Utils
      File "C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\Utils.py", line 10, in <module>
        from modules.sd_models_config import (find_checkpoint_config, config_default, config_sd2, config_sd2v, config_sd2_inpainting,
    ImportError: cannot import name 'find_checkpoint_config' from 'modules.sd_models_config' (C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\modules\sd_models_config.py)
```

---
```
*** Error loading script: CivitaihelperPrompts.py
*** Error loading script: Utils.py
*** Error loading script: batchCheckpointsPrompt.py
*** Error calling: C:\Users\user\Desktop\webui_forge_cu121_torch231\webui\extensions\sd-webui-segment-anything\scripts\sam.py/ui
```

---
```
2024-10-01 22:51:21,532 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\user\\Desktop\\stable-diffusion-webui\\models\\Stable-diffusion\\cyberrealistic_v50.safetensors', 'hash': 'd9a2cfca'}, 'additional_modules': [], 'unet_storage_dtype': None}
Using online LoRAs in FP16: False
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
IIB Database file has been successfully backed up to the backup folder.
```

1. But i dont know what is to do.
2. How to create a public link and where?

**Comment by DenOfEquity** [comment]
You've set Forge to use the A1111 virtual environment (venv), so it's using the older version of Torch that A1111 used. Best remove that `SET VENV=%A1111_HOME%/venv` line.
You can rename the main directory as you like.
Your models/loras/VAEs/embeddings/etc. are currently saved in the A1111 locations, so you can't simply delete the A1111 folder. Best to either move them into the directories Forge uses (layout is the same), or move to a common location that can be shared by any UIs you want to try.
Forge has its own controlnet, so it is correct that `sd-webui-controlnet` is unchecked in the Extensions tab.
Some of your old extensions will be incompatible with Forge2. check: https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/888 https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1754

**Comment by cxbignekoc** [comment]
Thx,

Do i need not modify the other ones?

 set A1111_HOME=C:/Users/user/Desktop/stable-diffusion-webui
 --ckpt-dir %A1111_HOME%/models/Stable-diffusion ^
 --hypernetwork-dir %A1111_HOME%/models/hypernetworks ^
 --embeddings-dir %A1111_HOME%/embeddings ^
 --lora-dir %A1111_HOME%/models/Lora

in user bat?

**Reply by DenOfEquity** [reply]
Have you moved your models/loras/etc.?

**Reply by cxbignekoc** [reply]
yes, i tried also to @REM all out and us RUN.BAT from main folder seems work well.

Looks now like this but it seems i dont need to use web-user.bat:
```

set PYTHON=C:/Users/user/AppData/Local/Programs/Python/Python310/python.exe
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --xformers

@REM Uncomment following code to reference an existing A1111 checkout.
@REM set A1111_HOME=C:/Users/user/Desktop/stable-diffusion-webui
@REM set VENV_DIR=%A1111_HOME%/venv
@REM set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% ^
@REM --ckpt-dir %A1111_HOME%/models/Stable-diffusion ^
@REM --hypernetwork-dir %A1111_HOME%/models/hypernetworks ^
@REM --embeddings-dir %A1111_HOME%/embeddings ^
@REM --lora-dir %A1111_HOME%/models/Lora
```

use of run.bat download magicprompt stuff again...

**Reply by DenOfEquity** [reply]
That looks good. You shouldn't need `set PYTHON=...` either.
If you still get the message about old xformers on startup then you can add ` --reinstall-xformers` to the `set COMMANDLINE_ARGS=...` line - but remember to remove it after next run.
When you use `run.bat`, it will then call `webui-user.bat`. So, using `run.bat` is correct.
